subprojects/core-api/src/main/java/org/gradle/api/component/CapabilityDescriptor.java

```diff
@@ -13,13 +13,28 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.gradle.internal.component.external.model;
+package org.gradle.api.component;
```

Comment:
I'm not sure this is the right place for this type. I'd put this, and `ComponentWithCapabilities` somewhere closer to dependency management.